R/http.R

```diff
@@ -646,7 +646,7 @@ httpTrace <- function(method, path, time) {
 }
 
 httpFunction <- function() {
-  httpType <- getOption("rsconnect.http", "rcurl")
+  httpType <- getOption("rsconnect.http", "curl")
```

Comment:
The curl method AFAICR is for using the external curl binary (on the mac) rather than the curl package. 

It's very easy to add a new method (we could add "curlpkg" or something like that) which stands alongside the other methods then we could gradually switch over to it as we become more comfortable).

Reply:
I'm not the "curl" method (which calls the curl binary not the curl package) is the correct default. I seem to recall issues with some of the streaming APIs. 

FWIW I maybe it makes sense to add support for the curl package or even httr.

Reply:
If that's the case I agree 100% we should move to using the `curl` package rather than `RCurl`.